axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -1000,5 +1020,27 @@ public io.axoniq.axonserver.connector.Registration registerSubscriptionQuery(Sub
                 return CompletableFuture.completedFuture(null);
             };
         }
+
+        private boolean awaitTermination(Duration timeout) {
+            Instant startAwait = Instant.now();
+            Instant endAwait = startAwait.plusSeconds(timeout.getSeconds());
+            while (Instant.now().isBefore(endAwait) && !queriesInProgress.isEmpty()) {
+                queriesInProgress.values()
+                                 .stream()
+                                 .findFirst()
+                                 .ifPresent(queryInProgress -> {
+                                     while (Instant.now().isBefore(endAwait) && queryInProgress.resultPending()) {
+                                         LockSupport.parkNanos(50);
```

Comment:
Since shutting down isn't time-critical, it might be better to wait a little longer. Perhaps 1ms? Or even 10? That way we don't risk this process taking CPU that's needed to finish tasks in progress

Reply:
Gotcha, this might be a bit "much" indeed; would be a waste to spend to many cycles on a busy-loop here. I'll go with 10ms in this case, as that's the most lenient option.